components/Modals/SignUp.tsx

```diff
@@ -36,6 +39,20 @@ export default function SignUpModal({ open, onClose }) {
 
   const handleSetPrivateKey = (pk: string) => {
     setPrivateKey(pk)
+    setCopyAcknowledged(true)
+  }
+
+  const copy = () => {
```

Comment:
We can use the  `navigator.clipboard.writeText` api for this. Think it also makes sense to move this to the utils folder so we can more easily reuse it later.